tool/loadgen/main.go

```diff
@@ -126,16 +148,32 @@ func test(containerName string) TestResult {
 	return tr
 }
 
+type Generator struct {
+	Url      string        `json:"url"`
+	Duration time.Duration `json:"duration"`
+	Rate     vegeta.Rate   `json:"rate"`
+}
+
+func NewGenerator(url string, duration time.Duration) Generator {
```

Comment:
It is not clear that we'll always use the same rate for all phases. Have you considered explicitly passing all options when creating a `Generator` value (i.e. no constructor function)?